lightning/src/ln/offers_tests.rs

```diff
@@ -391,6 +391,53 @@ fn prefers_more_connected_nodes_in_blinded_paths() {
 	}
 }
 
+/// Checks that an offer can be created with no blinded paths.
+#[test]
+fn create_offer_with_no_blinded_path() {
+	let chanmon_cfgs = create_chanmon_cfgs(2);
+	let node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+	let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+	let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+	create_announced_chan_between_nodes_with_value(&nodes, 0, 1, 10_000_000, 1_000_000_000);
+
+	let alice = &nodes[0];
+	let alice_id = alice.node.get_our_node_id();
+
+	let offer = alice.node
+		.create_offer_builder(None).unwrap()
+		.amount_msats(10_000_000)
+		.build().unwrap();
+	assert_eq!(offer.signing_pubkey(), Some(alice_id));
+	assert!(offer.paths().is_empty());
+}
+
+/// Checks that a refund can be created with no blinded paths.
+#[test]
+fn create_refund_with_no_blinded_path() {
+	let chanmon_cfgs = create_chanmon_cfgs(2);
+	let node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+	let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+	let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+	create_announced_chan_between_nodes_with_value(&nodes, 0, 1, 10_000_000, 1_000_000_000);
+
+	let alice = &nodes[0];
+	let alice_id = alice.node.get_our_node_id();
+
+	let absolute_expiry = Duration::from_secs(u64::MAX);
+	let payment_id = PaymentId([1; 32]);
+
+	let refund = alice.node
+		.create_refund_builder(None, 10_000_000, absolute_expiry, payment_id, Retry::Attempts(0), None)
+		.unwrap()
+		.build().unwrap();
+	assert_eq!(refund.amount_msats(), 10_000_000);
+	assert_eq!(refund.absolute_expiry(), Some(absolute_expiry));
+	assert_eq!(refund.payer_id(), alice_id);
+	assert!(refund.paths().is_empty());
+}
+
 /// Checks that blinded paths are compact for short-lived offers.
 #[test]
 fn creates_short_lived_offer() {
```

Comment:
The naming at least for the following tests is no longer relevant, it seems.

Reply:
As far as I understand, with the introduction of `BlindedPathParameters`, it seems like the behavior these tests were checking isn't relevant anymore. Should we think about removing these tests?

Reply:
Let's wait on the resolution of https://github.com/lightningdevkit/rust-lightning/pull/3246#discussion_r1745858783. We'll ultimately want to test cases allowed by whatever interface is exposed to the user.